src/metatrain/experimental/nanopet/modules/augmentation.py

```diff
@@ -118,17 +131,114 @@ def _apply_wigner_D_matrices(
+                                    system.types == int(key["center_type"]),
+                                )
+                            )
+                        )
+                elif (
+                    "first_atom" in block.samples.names
+                    and "second_atom" in block.samples.names
+                ):  # edge target
+                    for system in systems:
+                        neighbor_lists = system.known_neighbor_lists()
+                        if len(neighbor_lists) == 0:
+                            raise ValueError(
+                                "systems must have neighbor lists pre-computed. "
+                                "None found."
+                            )
+                        if len(neighbor_lists) > 1:
+                            raise ValueError("more than one neighbor list found")
+
+                        neighbor_samples = system.get_neighbor_list(
+                            neighbor_lists[0]
+                        ).samples
+                        split_indices.append(
+                            int(
+                                torch.sum(
+                                    torch.logical_and(
+                                        system.types[
+                                            neighbor_samples.column("first_atom")
+                                        ]
+                                        == int(key["first_atom_type"]),
+                                        system.types[
+                                            neighbor_samples.column("second_atom")
+                                        ]
+                                        == int(key["second_atom_type"]),
+                                    )
+                                )
+                            )
+                            // (2 if key["first_atom_type"] == key["second_atom_type"] else 1)  # as samples should be triangulated
+                        )
+                else:
+                    raise ValueError(f"unexpected samples: {block.samples}")
+
+            # TODO: implement rotations of 2 component axes
+            elif (
+                "o3_lambda_1" in target_tmap.keys.names
+                and "o3_lambda_2" in target_tmap.keys.names
+            ):  # two "o3_mu_x" components
+                if "atom" in block.samples.names:  # node target
+                    raise NotImplementedError(
+                        "rotations with 2 components axes not yet supported"
+                    )
+
+                elif (
+                    "first_atom" in block.samples.names
+                    and "second_atom" in block.samples.names
+                ):  # edge target
+                    raise NotImplementedError(
+                        "rotations with 2 components axes not yet supported"
+                    )
+
+                else:
+                    raise ValueError(f"unexpected samples: {block.samples}")
+
```

Comment:
Let's talk about this in person

Reply:
Removed all support for 2 components axes for now - outside the scope of this PR